src/predicates.ts

```diff
@@ -1182,3 +1195,23 @@ export class IsNotKubernetesQuantity implements Predicate<any> {
     };
   }
 }
+
+export class Not<T_El> implements Predicate<T_El> {
+  public static fromJson<E>(obj: any, registry: Registry): Not<E> {
+    return new Not<E>(registry.compile(obj.pred));
+  }
+
+  constructor(
+      private readonly pred: Predicate<T_El>,
+  ) {
+  }
+
+  public test(root: any): RuleMatchedAt {
+    const result = this.pred.test(root);
+    if (!result.matched && _.isUndefined(result.paths)) {
+      return { matched: !result.matched, paths: [""] };
```

Comment:
I think having an empty path here might cause unexpected things to happen. What about casting an empty array with `as`?

Reply:
or maybe just omit `paths` entirely?